topics/reference_guide/custom_language_support/navbar.md

```diff
@@ -0,0 +1,38 @@
+[//]: # "title: Navigation Bar"
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+The navigation bar implementation is used to customize and extend the [navigation bar](https://www.jetbrains.com/help/idea/guided-tour-around-the-user-interface.html#navigation-bar) structure.
+
+The starting point for the navigation bar extension is the [`NavBarModelExtension`](upsource:///platform/lang-impl/src/com/intellij/ide/navigationToolbar/NavBarModelExtension.java) interface, which is registered in the `com.intellij.navbar` extension point.
+
+To reuse the IntelliJ Platform implementation, you can extend one of two classes:
+
+- [`DefaultNavBarExtension`](upsource:///platform/lang-impl/src/com/intellij/ide/navigationToolbar/DefaultNavBarExtension.java)
+- [`StructureAwareNavBarModelExtension`](upsource:///platform/lang-impl/src/com/intellij/ide/navigationToolbar/StructureAwareNavBarModelExtension.java)
+
+## Default Navigation Bar
+
+`DefaultNavBarExtension` is the basic implementation of the navigation bar for any files. Inherit from this class if you want to create a simple navigation bar where you can change the display of folders or files for your language.
+
+In this case, you probably only need the following two methods to override:
+
+- `getPresentableText()` – returns a string representation of the navigation bar part element passed to it.
+- `getIcon()` – returns the icon for the element passed to it.
+
+## Structure Aware Navigation Bar
+
+`StructureAwareNavBarModelExtension` is an advanced implementation that provides the ability to display specific file elements in a bar. Such as, for example, the name of the class inside which the caret is currently located. Inherit from it if you want to add navigation bar support to your language with support for specific file elements.
+
+> Don't forget to implement [Structure View](structure_view.md) for correct work.
+{type="note"}
+
+In this case, you will also need to override the `getLanguage()` in addition to the two methods described earlier. This method returns the language instance for which this extension will work.
+
+The `adjustElement()` method allows you to modify the navigation bar element. It can be used, for example, when you want to show a class in the navigation bar when the caret is in a comment that is attached to the class.
```

Comment:
caret is in a -> caret is located in ...

Reply:
Fixed.